uat/driver/test_bundle.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+import os
+from pathlib import Path
+
+import pytest
+from lightkube import ApiError, Client, codecs
+from lightkube.generic_resource import create_global_resource, load_in_cluster_generic_resources
+from utils import assert_namespace_active, delete_job, fetch_job_logs, wait_for_job
+
+log = logging.getLogger(__name__)
+
+ASSETS_DIR = Path("assets")
+JOB_TEMPLATE_FILE = ASSETS_DIR / "test-job.yaml.j2"
+PROFILE_TEMPLATE_FILE = ASSETS_DIR / "test-profile.yaml.j2"
+
+TESTS_DIR = os.path.abspath(Path("integrations"))
+TESTS_IMAGE = "kubeflownotebookswg/jupyter-scipy:v1.7.0"
+
+NAMESPACE = "test-kubeflow"
+PROFILE_RESOURCE = create_global_resource(
+    group="kubeflow.org",
+    version="v1",
+    kind="profile",
+    plural="profiles",
+)
+
+JOB_NAME = "test-kubeflow"
+
+
+@pytest.fixture(scope="module")
+def lightkube_client():
+    """Initialise Lightkube Client."""
+    lightkube_client = Client()
+    load_in_cluster_generic_resources(lightkube_client)
+    return lightkube_client
+
+
+@pytest.fixture(scope="module")
+def create_profile(lightkube_client):
+    """Create Profile and handle cleanup at the end of the module tests."""
+    log.info(f"Creating Profile {NAMESPACE}...")
+    resources = list(
+        codecs.load_all_yaml(
+            PROFILE_TEMPLATE_FILE.read_text(),
+            context={"namespace": NAMESPACE},
+        )
+    )
+    assert len(resources) == 1, f"Expected 1 Profile, got {len(resources)}!"
+    lightkube_client.create(resources[0])
+
+    yield
+
+    # delete the Profile at the end of the module tests
+    log.info(f"Deleting Profile {NAMESPACE}...")
+    lightkube_client.delete(PROFILE_RESOURCE, name=NAMESPACE)
+
+
+@pytest.mark.abort_on_fail
+async def test_create_profile(lightkube_client, create_profile):
+    """Test Profile creation.
+
+    This test relies on the create_profile fixture, which handles the Profile creation and
+    is responsible for cleaning up at the end.
+    """
+    try:
+        profile_created = lightkube_client.get(
+            PROFILE_RESOURCE,
+            name=NAMESPACE,
+        )
+    except ApiError as e:
+        if e.status == 404:
+            profile_created = False
+        else:
+            raise
+    assert profile_created, f"Profile {NAMESPACE} not found!"
+
+    assert_namespace_active(lightkube_client, NAMESPACE)
+
+
+def test_kubeflow(lightkube_client):
```

Comment:
Maybe something more descriptive here.  `kubeflow` is quite broad.  `test_kubeflow_workloads`?  I'm not sure.  work `e2e` in somehow?

It reads awkwardly since it must have `test` at the start.  Ideally its maybe `run_e2e_workload_tests` or similar

Reply:
Well, I could tweak the test discovery naming convention for the driver, but I'm not terribly excited about it: https://docs.pytest.org/en/7.1.x/example/pythoncollection.html#changing-naming-conventions

We could do `test_kubeflow_e2e_workflows` bringing your suggestions together.

Reply:
yeah, lets use the awkward language rather than changing the discovery :D